File: src/test/java/test/CreateRoleTest.java
import cn.authing.sdk.java.client.ManagementClient;
import cn.authing.sdk.java.dto.*;
import cn.authing.sdk.java.model.ManagementClientOptions;
import cn.authing.sdk.java.util.JsonUtils;

import java.util.Collections;



public class CreateRoleTest {

    private static final String ACCESS_KEY_ID = "YOUR_ACCESS_KEY_ID";
    private static final String ACCESS_KEY_SECRET = "YOUR_ACCESS_KEY_SECRET";


    public static void main(String[] args) throws Throwable {
        ManagementClientOptions clientOptions = new ManagementClientOptions(ACCESS_KEY_ID, ACCESS_KEY_SECRET);
        ManagementClient client = new ManagementClient(clientOptions);

        CreateRoleDto request = new CreateRoleDto();
        request.setCode("code_7534");
        request.setNamespace("namespace_3017");
        request.setDescription("description_516");

        RoleSingleRespDto response = client.createRole(request);
        System.out.println(JsonUtils.serialize(response));
    }

}

File: src/test/go/test/CreateRoleTest.go
package main

import (
"authing-go-sdk/client"
"authing-go-sdk/dto"
"fmt"
)

func main() {
    options := client.ManagementClientOptions{
        AccessKeyId:     "AUTHING_USERPOOL_ID",
        AccessKeySecret: "AUTHING_USERPOOL_SECRET",
    }
    var err error
    client, err := client.NewClient(&options)
    if err != nil {
        panic(err)
    }
    request := dto.CreateRoleDto{
        Code: "code_9045",
        Namespace: "namespace_3497",
        Description: "description_4139",
    }
    response := client.CreateRole(&request)
    fmt.Println(response)
}

